src/Analyzer.TemplateProcessor/ArmTemplateProcessor.cs

```diff
@@ -392,58 +391,47 @@ private TemplateExpressionEvaluationHelper GetTemplateFunctionEvaluationHelper(T
-        /// <returns>An array of the Template Resources after they have been reordered.</returns>
-        private TemplateResource[] ReorderResourceCopies(Template template, Dictionary<string, (string, int)> copyNameMap)
+        /// <returns>An array of the Template Resources after they have been dropped if needed.</returns>
+        private TemplateResource[] MapResources(Template template, Dictionary<string, (string, int)> copyNameMap)
         {
             // Set OriginalName back on resources that were copied and reorder the resources.
             // Omit extra copies of resources if needed.
-            List<TemplateResource> updatedOrder = new List<TemplateResource>();
+            List<TemplateResource> resources = new List<TemplateResource>();
             for (int i = 0; i < template.Resources.Length; i++)
             {
                 var resource = template.Resources[i];
-                if (resource.Copy == null)
+                if (resource.Copy != null)
                 {
-                    // non-copied resource.  Add to updated array
-                    updatedOrder.Add(resource);
-                }
-                else
-                {
-                    // Copied resource.  Update OriginalName and:
-                    // - if it's the first copy, insert back where it was supposed to be
-                    // - if it's an extra copy, add to the end, or don't add at all if requested
+                    // Copied resource.  Update OriginalName and
+                    // add maping to original resource
                     if (copyNameMap.TryGetValue(resource.Copy.Name.Value, out (string, int) originalValues))
                     {
                         resource.OriginalName = originalValues.Item1;
-                        if (resource.CopyContext.CopyIndex == 0)
-                        {
-                            updatedOrder.Insert(originalValues.Item2, resource);
-                        }
-                        else if (!dropResourceCopies)
-                        {
-                            updatedOrder.Add(resource);
-                            resource.Path = $"resources[{updatedOrder.Count - 1}]";
-                        }
 
-                        if (!dropResourceCopies)
+                        if (!dropResourceCopies || resource.CopyContext.CopyIndex == 0)
                         {
+                            resources.Add(resource);
+                            resource.Path = $"resources[{resources.Count - 1}]";
                             AddResourceMapping(resource.Path, $"resources[{originalValues.Item2}]");
                         }
-                    }
-                    else
-                    {
-                        // Couldn't get original values.  Insert at end as a precaution,
-                        // but this code should never be reached under normal circumstances.
-                        updatedOrder.Add(resource);
+
+                        continue;
```

Comment:
It looks like this only skips one line of code.  That line should be moved into the 'else' block so `continue` can be removed.